### ISSUE TITLE
How to know what profiles are activated with each task?

### ISSUE THREAD

**Issue by qianyan** [technomancy/leiningen#2153]
> Leiningen will automatically merge the :dev profile into the project configuration for the repl, test, and run tasks, but not jar or uberjar.

from this quoted comment, I am really confused how to know what profiles are activated by default for each task? for example: if I run `lein run`, what profiles are activated?

**Comment by hyPiRion** [comment]
Yeah, this is poorly documented, and is often messes with the developer's expectations.

I think we should write it up, but I'm interested in getting this way simpler for 3.0.

**Comment by LukeDefeo** [comment]
I also have this confusion, some better docs in this area would be really helpful.

**Comment by LukeDefeo** [comment]
It would nice it lein printed out the list of profiles available and then also the profiles its actually merged for the task. Maybe this could be a verbose option
